### PR TITLE
[no ticket][risk=no] Refactor calls to get chart data

### PR DESCRIPTION
**Pull request by chenchals**
Description:
---
- Refactored all calls to get chart data for showing plots/graphs
- No API calls were changed

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [x] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+  public void addWhereClause(
+      ParticipantCriteria participantCriteria,
+      String mainTable,
+      StringBuilder queryBuilder,
+      Map<String, QueryParameterValue> params) {
+    SearchRequest request = participantCriteria.getSearchRequest();
+    if (request == null) {
+      queryBuilder.append(
+          QueryBuilder.PERSON_ID_WHITELIST_TEMPLATE.replace("${mainTable}", mainTable));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:41]:
```suggestion
          PERSON_ID_WHITELIST_TEMPLATE.replace("${mainTable}", mainTable));
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      params.put(
+          QueryBuilder.PERSON_ID_WHITELIST_PARAM,
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:43]:
```suggestion
          PERSON_ID_WHITELIST_PARAM,
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          QueryParameterValue.array(
+              participantCriteria.getParticipantIdsToInclude().toArray(new Long[0]), Long.class));
+    } else {
+      if (request.getIncludes().isEmpty() && request.getExcludes().isEmpty()) {
+        QueryBuilder.log.log(
+            Level.WARNING, "Invalid SearchRequest: includes[] and excludes[] cannot both be empty");
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:49]:
```suggestion
        log.log(
            Level.WARNING, "Invalid SearchRequest: includes[] and excludes[] cannot both be empty");
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+      // if includes is empty then don't add the excludes clause
+      if (joiner.toString().isEmpty()) {
+        joiner.merge(buildQuery(request.getExcludes(), mainTable, params, false));
+      } else {
+        joiner.merge(buildQuery(request.getExcludes(), mainTable, params, true));
+      }
+      Set<Long> participantIdsToExclude = participantCriteria.getParticipantIdsToExclude();
+      if (!participantIdsToExclude.isEmpty()) {
+        joiner.add(QueryBuilder.PERSON_ID_BLACKLIST_TEMPLATE.replace("${mainTable}", mainTable));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:65]:
```suggestion
        joiner.add(PERSON_ID_BLACKLIST_TEMPLATE.replace("${mainTable}", mainTable));
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        params.put(
+            QueryBuilder.PERSON_ID_BLACKLIST_PARAM,
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:67]:
```suggestion
            PERSON_ID_BLACKLIST_PARAM,
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      Map<String, QueryParameterValue> params,
+      Boolean excludeSQL) {
+    StringJoiner joiner = new StringJoiner("AND ");
+    List<String> queryParts = new ArrayList<>();
+    for (SearchGroup includeGroup : groups) {
+      SearchGroupItemQueryBuilder.buildQuery(params, queryParts, includeGroup);
+
+      if (excludeSQL) {
+        joiner.add(
+            QueryBuilder.EXCLUDE_SQL_TEMPLATE
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:99]:
```suggestion
            EXCLUDE_SQL_TEMPLATE
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                .replace("${mainTable}", mainTable)
+                .replace("${excludeSql}", String.join(QueryBuilder.UNION_TEMPLATE, queryParts)));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:101]:
```suggestion
                .replace("${excludeSql}", String.join(UNION_TEMPLATE, queryParts)));
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+      } else {
+        joiner.add(
+            QueryBuilder.INCLUDE_SQL_TEMPLATE
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:104]:
```suggestion
            INCLUDE_SQL_TEMPLATE
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+                .replace("${mainTable}", mainTable)
+                .replace("${includeSql}", String.join(QueryBuilder.UNION_TEMPLATE, queryParts)));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:106]:
```suggestion
                .replace("${includeSql}", String.join(UNION_TEMPLATE, queryParts)));
```

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortQueryBuilder.java [hunk 10]
```diff
@@ -81,25 +36,12 @@ public class CohortQueryBuilder {
...
-
-  private CBCriteriaDao cbCriteriaDao;
+  //  private CBCriteriaDao cbCriteriaDao;
   private static final Logger log = Logger.getLogger(CohortQueryBuilder.class.getName());
 
   @Autowired
-  public CohortQueryBuilder(CBCriteriaDao cbCriteriaDao) {
-    this.cbCriteriaDao = cbCriteriaDao;
+  public CohortQueryBuilder() {
+    // this.cbCriteriaDao = cbCriteriaDao;
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortQueryBuilder.java]: I'm assuming this line should be removed?

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+
+  /**
+   * Provides counts with demographic info for charts defined by the provided {@link
+   * ParticipantCriteria}.
+   */
+  public QueryJobConfiguration buildDemoChartInfoCounterQuery(
+      ParticipantCriteria participantCriteria) {
+    Map<String, QueryParameterValue> params = new HashMap<>();
+    String sqlTemplate =
+        ChartQueryBuilder.DEMO_CHART_INFO_SQL_TEMPLATE
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:94]:
```suggestion
        DEMO_CHART_INFO_SQL_TEMPLATE
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+            .replace("${genderOrSex}", participantCriteria.getGenderOrSexType().toString())
+            .replace(
+                "${ageRange1}",
+                ChartQueryBuilder.getAgeRangeSql(18, 44, participantCriteria.getAgeType()))
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:98]:
```suggestion
                getAgeRangeSql(18, 44, participantCriteria.getAgeType()))
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+            .replace(
+                "${ageRange2}",
+                ChartQueryBuilder.getAgeRangeSql(45, 64, participantCriteria.getAgeType()));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:101]:
```suggestion
                getAgeRangeSql(45, 64, participantCriteria.getAgeType()));
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+    StringBuilder queryBuilder = new StringBuilder(sqlTemplate);
+    addWhereClause(
+        participantCriteria, ChartQueryBuilder.SEARCH_PERSON_TABLE, queryBuilder, params);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:104]:
```suggestion
        participantCriteria, SEARCH_PERSON_TABLE, queryBuilder, params);
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+    addDataFilters(participantCriteria.getSearchRequest().getDataFilters(), queryBuilder, params);
+    queryBuilder.append(ChartQueryBuilder.DEMO_CHART_INFO_SQL_GROUP_BY);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:106]:
```suggestion
    queryBuilder.append(DEMO_CHART_INFO_SQL_GROUP_BY);
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+  }
+
+  /**
+   * Provides counts with ethnicity info for cohort defined by the provided {@link
+   * ParticipantCriteria}.
+   */
+  public QueryJobConfiguration buildEthnicityInfoCounterQuery(
+      ParticipantCriteria participantCriteria) {
+    Map<String, QueryParameterValue> params = new HashMap<>();
+    StringBuilder queryBuilder = new StringBuilder(ChartQueryBuilder.ETHNICITY_INFO_SQL_TEMPLATE);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:121]:
```suggestion
    StringBuilder queryBuilder = new StringBuilder(ETHNICITY_INFO_SQL_TEMPLATE);
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+    addWhereClause(
+        participantCriteria, ChartQueryBuilder.SEARCH_PERSON_TABLE, queryBuilder, params);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:123]:
```suggestion
        participantCriteria, SEARCH_PERSON_TABLE, queryBuilder, params);
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 18]
```diff
@@ -0,0 +1,192 @@
...
+    addDataFilters(participantCriteria.getSearchRequest().getDataFilters(), queryBuilder, params);
+    queryBuilder.append(ChartQueryBuilder.ETHNICITY_INFO_SQL_GROUP_BY);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:125]:
```suggestion
    queryBuilder.append(ETHNICITY_INFO_SQL_GROUP_BY);
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 19]
```diff
@@ -0,0 +1,192 @@
...
+        .setUseLegacySql(false)
+        .build();
+  }
+  /**
+   * Provides counts with domain info for charts defined by the provided {@link
+   * ParticipantCriteria}.
+   */
+  public QueryJobConfiguration buildDomainChartInfoCounterQuery(
+      ParticipantCriteria participantCriteria, Domain domain, int chartLimit) {
+    StringBuilder queryBuilder = new StringBuilder(ChartQueryBuilder.ID_SQL_TEMPLATE);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:150]:
```suggestion
    StringBuilder queryBuilder = new StringBuilder(ID_SQL_TEMPLATE);
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 20]
```diff
@@ -0,0 +1,192 @@
...
+    Map<String, QueryParameterValue> params = new HashMap<>();
+    addWhereClause(
+        participantCriteria, ChartQueryBuilder.SEARCH_PERSON_TABLE, queryBuilder, params);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:153]:
```suggestion
        participantCriteria, SEARCH_PERSON_TABLE, queryBuilder, params);
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 21]
```diff
@@ -0,0 +1,192 @@
...
+    addDataFilters(participantCriteria.getSearchRequest().getDataFilters(), queryBuilder, params);
+    String searchPersonSql = queryBuilder.toString();
+    queryBuilder =
+        new StringBuilder(
+            ChartQueryBuilder.DOMAIN_CHART_INFO_SQL_TEMPLATE.replace(
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:158]:
```suggestion
            DOMAIN_CHART_INFO_SQL_TEMPLATE.replace(
```

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java [hunk 22]
```diff
@@ -0,0 +1,192 @@
...
+                "${innerSql}", searchPersonSql));
+    String paramName =
+        QueryParameterUtil.addQueryParameterValue(
+            params, QueryParameterValue.string(domain.name()));
+    String endSqlTemplate =
+        ChartQueryBuilder.DOMAIN_CHART_INFO_SQL_GROUP_BY
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java:164]:
```suggestion
        DOMAIN_CHART_INFO_SQL_GROUP_BY
```

**Review by freemabd — COMMENTED** [review]
This all seems fine. Left some suggestions about code cleanup. 

Just a few thoughts on the packages. So currently we have 5 packages where all data apps code is located:
1. org.pmiops.workbench.cohortbuilder
2. org.pmiops.workbench.cohortreview
3. org.pmiops.workbench.cohorts
4. org.pmiops.workbench.conceptset
5. org.pmiops.workbench.dataset

I would prefer to keep the chart classes in a package within these 5 packages. So instead of org.pmiops.workbench.chart maybe create another package org.pmiops.workbench.cohortbuilder.chart and move ChartQueryBuilder, ChartService and ChartServiceImpl into it.

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 1 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:41]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 2 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:43]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 3 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:49]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 4 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:65]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 5 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:67]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 6 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:99]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 7 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:101]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 8 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:104]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java (hunk 9 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilder.java:106]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 18 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 19 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 20 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 21 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 22 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortQueryBuilder.java (hunk 10 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortQueryBuilder.java]: Yes. cbCriteriaDao is not used

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 11 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 12 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 13 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 14 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 15 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 16 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

File: api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java (hunk 17 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/chart/ChartQueryBuilder.java]: done

**Pull request merged**